packages/contracts/src/core/dao/DAO.sol

```diff
@@ -320,25 +323,30 @@ contract DAO is
     }
 
     /// @inheritdoc IDAO
-    function setSignatureValidator(
-        address _signatureValidator
-    ) external override auth(SET_SIGNATURE_VALIDATOR_PERMISSION_ID) {
-        signatureValidator = IERC1271(_signatureValidator);
-
-        emit SignatureValidatorSet({signatureValidator: _signatureValidator});
+    function setSignatureValidator(address) external pure override {
+        revert FunctionRemoved();
     }
 
     /// @inheritdoc IDAO
+    /// @dev Relays the validation logic determining who is allowed to sign on behalf of the DAO to its permission manager.
+    /// Caller specific bypassing can be set direct granting (i.e., `grant({_where: dao, _who: specificErc1271Caller, _permissionId: VALIDATE_SIGNATURE_PERMISSION_ID})`).
+    /// Caller specific signature validation logic can be set by granting with a `PermissionCondition` (i.e., `grantWithCondition({_where: dao, _who: specificErc1271Caller, _permissionId: VALIDATE_SIGNATURE_PERMISSION_ID, _condition: yourConditionImplementation})`)
+    /// Generic signature validation logic can be set for all calling contracts by granting with a `PermissionCondition` to `PermissionManager.ANY_ADDR()` (i.e., `grantWithCondition({_where: dao, _who: PermissionManager.ANY_ADDR(), _permissionId: VALIDATE_SIGNATURE_PERMISSION_ID, _condition: yourConditionImplementation})`).
     function isValidSignature(
         bytes32 _hash,
         bytes memory _signature
     ) external view override(IDAO, IERC1271) returns (bytes4) {
-        if (address(signatureValidator) == address(0)) {
-            // Return the invalid magic number
-            return bytes4(0);
+        if (
+            isGranted({
+                _where: address(this),
+                _who: msg.sender,
+                _permissionId: VALIDATE_SIGNATURE_PERMISSION_ID,
+                _data: abi.encode(_hash, _signature)
+            })
+        ) {
+            return 0x1626ba7e; // `type(IERC1271).interfaceId` = bytes4(keccak256("isValidSignature(bytes32,bytes)")`
         }
-        // Forward the call to the set signature validator contract
-        return signatureValidator.isValidSignature(_hash, _signature);
+        return 0xffffffff; // `bytes4(uint32(type(uint32).max-1))`
```

Comment:
maybe an alternative to give the decision only to the more specific condition and don't fall back to the more generic one in case the specific condition returns false
```solidity
       address accessFlagOrCondition = permissionsHashed[
            permissionHash(address(this), msg.sender, VALIDATE_SIGNATURE_PERMISSION_ID)
        ];

        if (accessFlagOrCondition == ALLOW_FLAG) return 0x1626ba7e;
        if (accessFlagOrCondition == UNSET_FLAG) {
            if(_isGranted(address(this), ANY_ADDR, VALIDATE_SIGNATURE_PERMISSION_ID, abi.encode(_hash, _signature))) {
                return 0x1626ba7e;
            }
            return 0xffffffff;
        };
        if(_isGranted(address(this), msg.sender, VALIDATE_SIGNATURE_PERMISSION_ID, abi.encode(_hash, _signature))) {
            return 0x1626ba7e;
        }

        return 0xffffffff;
        ```

Reply:
I agree, we should take more time to discuss this fallback. This seems safer to me.

Reply:
will be realized in #461